feat(compiler-cli): Add an extended diagnostic for `ngSkipHydration` (#49512)
angular/angular: aio/content/extended-diagnostics/NG8108.md

> @name ngSkipHydration biniding

Reviewer: ```suggestion
@name ngSkipHydration should be a static attribute
```

@description

> This diagnostic ensures that the special attribute `ngSkipHydration` is not a binding and has no 
> other value than `"true"` or an empty value. 
> 
> 

Reviewer: ```suggestion
The `ngSkipHydration` is a special attribute that indicates to Angular that a particular component should be
opted-out of hydration. This diagnostic ensures that this attribute `ngSkipHydration` is set statically and the 
value is either set to `"true"` or an empty value.

```

## What should I do instead?

> When using `ngSkipHydration`, ensure you do not use the Angular template binding syntax.

Reviewer: ```suggestion
When using the `ngSkipHydration`, ensure that it's set as a static attribute (i.e. you do not use the Angular template binding syntax).
```

> <pre>
>   <my-cmp ngSkipHydration></my-cmp>
>   <!-- or -->
>   <my-cmp ngSkipHydration="true"></my-cmp>
> </pre>

Reviewer: We can use the same approach as in other examples:
```suggestion
<code-example format="html" language="html">
&lt;my-cmp ngSkipHydration /&gt;
&lt;!-- or --&gt;
&lt;my-cmp ngSkipHydration="true" /&gt;
</code-example>
```
